### DISCUSSION TITLE
custom menu

### DISCUSSION THREAD

**Discussion by sandokan815** [algolia/instantsearch#5287] [Q&A] [answered]
HI, Could you let me know how can I add icon or images in menu?
As a default, react-instantsearch supports only text-basis menu, but I want to add icons like this.
![image](https://user-images.githubusercontent.com/35192503/136248488-ef01c0e7-91dd-4ba0-af48-4570408dd7c7.png)

**Comment by francoischalifour** [comment, marked as answer]
Hey! Algolia returns the facets of your records as text, so you need to map each label to its image in the frontend using [`connectMenu`](https://www.algolia.com/doc/api-reference/widgets/menu/react/#connector).
